test/UnitTests/TestFramework.UnitTests/Assertions/CollectionAssertTests.cs

```diff
@@ -215,6 +215,38 @@ public void CollectionAssertAreEqualComparerMessageParametersNullabilityPostCond
         comparer.ToString(); // no warning
     }
 
+    public void CollectionAssertAreEqual_EqualNestedLists_Passes()
```

Comment:
Can you please add a negative test for both AreEqual and AreNotEqual (a test that fails)